### PR TITLE
Adding localuser and user flags

### PR DESCRIPTION
**Pull request by holloway**
This adds two flags to the CLI interface,

1. `-l/--localuser` to get the Git username/email from the local repo config rather than global.
2. `-u/--username <username,email>` to set the username/email.

### CODE REVIEW

File: bin/gh-pages [hunk 1]
```diff
@@ -39,32 +40,57 @@ function main(args) {
       '.'
     )
     .option('-n, --no-push', 'Commit only (with no push)')
+    .option("-l, --localuser", "Use Git local user config (not global)")
+    .option("-u, --user <username,email>", "Override default username, email")
     .parse(args);
 
-  ghpages.publish(
```
Comment (holloway) [inline bin/gh-pages]: this was moved to a `publish` function below

File: bin/gh-pages [hunk 2]
```diff
@@ -39,32 +40,57 @@ function main(args) {
...
+    ]).then(results => {
+      publish({
+        name: results[0].output.trim(),
+        email: results[1].output.trim()
+      });
+    });
+  } else if (program.user) {
+    var userParts = program.user.split(",");
+    publish({
+      user: userParts[0],
```
Comment (saiichihashimoto) [inline bin/gh-pages]: [Above](https://github.com/tschaub/gh-pages/pull/229/files#diff-2a9efa0b08eba0bbe5896df72f13afd6R53), this key is `name`, not `user`.

Comment (holloway) [reply bin/gh-pages]: Cheers, i'll have a look at this in the next few days

**Comment by saiichihashimoto** [comment]
@tschaub, this seems to work! Any idea on what the ETA for this will be?

**Comment by just-a-bad-day** [comment]
Any updates? I'd really like to have this feature

**Comment by saiichihashimoto** [comment]
image

@just-a-bad-day, I wouldn't hold my breath...

**Comment by holloway** [comment]
I believe this gh-pages package is mentioned in the Create-React-App docs, so if it's not maintained then we should probably tell them to remove it.

**Comment by tschaub** [comment]
@holloway - thanks for your interest in contributing.

It looks like you prefer `"` to `'`.  I hope we agree that the difference is not significant, but that consistency with existing conventions is good.  The linter should have caught this, but the `bin` files were being ignored.  I fixed this in #239.

Please rebase your work and remove the unnecessary parts of the change (all the changes from `'` to `"`).  In general, it is nice to see pull requests that don't include extraneous whitespace or formatting changes - making it easier for a reviewer to focus on the important part.

It would also be nice to see a test for this change.  It should be straightforward to at least add [a scenario](https://github.com/tschaub/gh-pages/blob/master/test/bin/gh-pages.spec.js#L27-L34) that confirms the `user` option is properly set given your new arguments.

Two other things that come to mind regarding this change:

1. Rather than adding two new arguments, it could make sense to change the behavior so that the "local" user is used by default (breaking change), and then add the `user` option to override.

2. I think more people might expect [RFC-5322](https://tools.ietf.org/html/rfc5322#section-3.4) address style (e.g. `"Author Name <email@address.com>"`).  This would make it similar, for example, to the [`author` option](https://git-scm.com/docs/git-commit#git-commit---authorltauthorgt) for `git commit`.

Curious to hear what you think.

**Comment by nuklearfiziks** [comment]
This PR is really stale but I want to see it happen. I've rebased and fixed the linting errors here:

https://github.com/nuklearfiziks/gh-pages/tree/feature/user-cli

Local user should be used by default in all instances because that's the one Git gives priority to (AFAICT); fixing that doesn't seem like a breaking change so much as a bugfix TBTH.

Agreed default should be the priority Git uses, followed by a flag override; also re: supplying it as close to Git syntax as possible (i.e., RFC-5322).

**Comment by tschaub** [comment]
Instead of adding any new options, I've changed how the git user is determined.  See #264 for detail.  If running `git config user.name` and `git config user.email` in the current working directory does not return the correct user, the `user` option can be used.  This was already available in the API, and #264 exposes it to the CLI as well.  The format of the option for the CLI is `Your Name <email@example.com`.

Thanks for the work on this @holloway and @nuklearfiziks.

**Pull request closed without merging**